hazelcast/src/main/java/com/hazelcast/nio/tcp/AbstractIOSelector.java

```diff
@@ -102,7 +102,16 @@ private void processSelectionQueue() {
             if (runnable == null) {
                 return;
             }
-            runnable.run();
+            if (runnable instanceof WriteHandler) {
```

Comment:
Can you add a few lines of comment what is being done here for future generations? I'm sure people will get confused.